### PR TITLE
chore: Sunset Kong Gateway versions 2.1.x-2.5.x

### PR DESCRIPTION
**Pull request by lena-larionova**
### Description

As of August 24th, 2023, Kong Gateway OSS and Enterprise versions 2.1.x-2.5.x [end their sunset support](https://docs.konghq.com/gateway/latest/support-policy/#older-versions).

This PR removes all of the old versions from the docs site, as well as the `/archive` folder maintained on the `main` branch.
We no longer need that folder, since we have a legacy-gateway site.

You can find all of the archived versions at the following links:
https://legacy-gateway--kongdocs.netlify.app/enterprise/2.5.x/
https://legacy-gateway--kongdocs.netlify.app/gateway-oss/2.5.x/
https://legacy-gateway--kongdocs.netlify.app/getting-started-guide/2.5.x/overview/

For the additions to the archive, see https://github.com/Kong/docs.konghq.com/pull/5993.

https://konghq.atlassian.net/browse/DOCU-3347

---

Some specifics:
* Deleting everything to do with `enterprise`, `gateway-oss`, and `getting-started-guide` editions and URLs. All three of those editions were replaced with one `gateway` edition in 2.6.x. Since we're archiving everything before 2.6.x, we no longer need to maintain all of the exceptions for the old editions.
* Plugins: We will no longer build versions 2.5.x or earlier. There wasn't a lot of 2.5.x or earlier content that was unique, so I think that should be fine.
  * If we want to move the old plugin docs to the archive site, we will need to port the whole new plugin hub over there. I'm not sure that it's worth it yet.

### Testing instructions

Netlify link: https://deploy-preview-5992--kongdocs.netlify.app/gateway/latest/

### Checklist 

- [x] Review label added 
- [x] PR pointed to correct branch (`main` for immediate publishing, or a release branch: e.g. `release/gateway-3.2`, `release/deck-1.17`)

### CODE REVIEW

**Comment by lena-larionova** [comment]
@fabianrbz I think we have to rewrite a bunch of tests to use Vault Authentication instead of JWT Signer, since Vault has a different source file up to version 2.7.x and JWT Signer stops at 2.5.x. Is there a way to generate these specs/tests, or is it a manual change?

**Comment by fabianrbz** [comment]
@lena-larionova I've fixed the specs. They actually use a [fixture app](https://github.com/Kong/docs.konghq.com/tree/main/spec/fixtures/app/) that only includes a subset of plugins with dummy content. I've updated those to use the new versions only.

File: app/_hub/kong-inc/jwt-signer/_2.2.x/_index.md [hunk 1]
```diff
@@ -14,9 +14,7 @@ can be any valid verifiable tokens. E.g. two access tokens
 * [Plugin Configuration Parameters](#plugin-configuration-parameters)
   * [Description of Plugin Configuration Parameters](#description-of-plugin-configuration-parameters)
     * [config.realm](#configrealm)
```
Comment (fabianrbz) [inline app/_hub/kong-inc/jwt-signer/_2.2.x/_index.md:16]: I think we can safely remove the whole `app/_hub/kong-inc/jwt-signer/_2.2.x/` folder

File: app/_includes/md/2.5.x/deployment-options-k8s.md [hunk 2]
```diff
@@ -1,4 +0,0 @@
-<!-- Deployment Options section; used in all Enterprise k8s installation topics-->
-The following instructions assume that you are deploying {{site.base_gateway}} in [classic embedded mode](/enterprise/{{include.kong_version}}/deployment/deployment-options/).
-
-If you would like to run {{site.base_gateway}} in Hybrid mode, the instructions in this topic will walk you though setting up a Control Plane instance. Afterward, you will need to bring up additional gateway instances for the Data Planes, and perform further configuration steps. See [Hybrid Mode setup documentation](https://github.com/Kong/charts/blob/main/charts/kong#hybrid-mode) for details.
```
Comment (fabianrbz) [inline app/_includes/md/2.5.x/deployment-options-k8s.md:4]: This file is referenced in the `Readme.md`, we probably need to update it to use something else

File: app/_redirects [hunk 3]
```diff
@@ -887,16 +820,18 @@
...
-/2.5.x/*                                    /gateway-oss/2.5.x/:splat
-/2.4.x/*                                    /gateway-oss/2.4.x/:splat
-/2.3.x/*                                    /gateway-oss/2.3.x/:splat
-/2.2.x/*                                    /gateway-oss/2.2.x/:splat
-/2.1.x/*                                    /gateway-oss/2.1.x/:splat
+/enterprise/2.1.x/*     https://legacy-gateway--kongdocs.netlify.app/enterprise/2.1.x/:splat
+/enterprise/2.2.x/*     https://legacy-gateway--kongdocs.netlify.app/enterprise/2.2.x/:splat
+/enterprise/2.3.x/*     https://legacy-gateway--kongdocs.netlify.app/enterprise/2.3.x/:splat
+/enterprise/2.4.x/*     https://legacy-gateway--kongdocs.netlify.app/enterprise/2.4.x/:splat
+/enterprise/2.5.x/*     https://legacy-gateway--kongdocs.netlify.app/enterprise/2.5.x/:splat
```
Comment (fabianrbz) [inline app/_redirects:828]: do we need to add redirects for the old getting-started-guides?

**Comment by lena-larionova** [comment]
@fabianrbz and it just doubled, with the removal of the archive 😁 
Screenshot 2023-08-23 at 10 20 39 AM

**Pull request merged**